frontend/src/components/user-administration/user-administration-modal/user-administration-modal.tsx

```diff
@@ -0,0 +1,278 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+  Box,
+  Button,
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Modal,
+  Select,
+  SelectChangeEvent,
+  Typography
+} from '@mui/material';
+import React, { useState } from 'react';
+import { TextFieldElement, useForm } from 'react-hook-form-mui';
+import { toast } from 'react-toastify';
+import { z, ZodIssueOptionalMessage, ErrorMapCtx } from 'zod';
+import { makeStyles } from '@mui/styles';
+import { Icon } from '@iconify/react';
+import { User } from 'pages/api/user';
+import ShowPasswordModal from './show-password-modal';
+import { UserDto } from 'src/models/user/user.model';
+
+const UserAdministrationModal = ({
+  showModal = true,
+  setShowModal,
+  onAddedUser,
+  user
+}: {
+  showModal: boolean;
+  setShowModal: (value: boolean) => void;
+  onAddedUser: (value: any) => void;
+  user: User;
+}) => {
+  const formSchema = z.object({
+    firstname: z.string().min(3),
+    lastname: z.string().min(1),
+    email: z.string().email()
+  });
```

Comment:
You can interfier a interface from the schema.

```suggestion
  const formSchema = z.object({
    firstname: z.string().min(3),
    lastname: z.string().min(1),
    email: z.string().email()
  });
  
  export type CreateUserType = z.infer<typeof formSchema>;

Then you can use that aswell if needed
  
```